trio/_threads.py

```diff
@@ -422,6 +427,13 @@ def from_thread_run_sync(fn, *args, trio_token=None):
           "foreign" thread, spawned using some other framework, and still want
           to enter Trio.
     """
+
+    if not callable(fn) or inspect.iscoroutinefunction(fn):
```

Comment:
I actually don't think you need to explicitly check whether it's callable, because the TypeError you get from calling a non-callable will properly propagate back into the caller of `trio.from_thread.run_sync` when you try to call the function below.

As I explained in chat, `inspect.iscoroutinefunction` will not catch all async callables. That's OK here, but you will give a better user experience (ie, catch more problem cases) if you wait until after you've called the function, and then throw an error if the value you got back from it satisfies `inspect.iscoroutine`.